shared/teams/team/members/container.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as I from 'immutable'
+import {type TypedState, connect} from '../../../util/container'
+import {Members} from '.'
+
+export type OwnProps = {
+  teamname: string,
+}
+
+const order = {owner: 0, admin: 1, writer: 2, reader: 3}
+
+const getOrderedMemberArray = (
+  memberInfo: I.Set<Types.MemberInfo>,
+  you: ?string,
+  listYouFirst: boolean
+): Array<Types.MemberInfo> => {
+  let youInfo
+  let info = memberInfo
+  if (you && !listYouFirst) {
+    youInfo = memberInfo.find(member => member.username === you)
+    if (youInfo) {
+      info = memberInfo.delete(youInfo)
+    }
+  }
+  let returnArray = info
+    .toArray()
+    .sort(
+      (a, b) =>
+        !a.type || !b.type || a.type === b.type
+          ? a.username.localeCompare(b.username)
+          : order[a.type] - order[b.type]
+    )
+
+  if (youInfo) {
+    returnArray.unshift(youInfo)
+  }
+  return returnArray
+}
+
+const mapStateToProps = (state: TypedState, {teamname}: OwnProps) => ({
+  // Assume teamname exists here because parent throws an error if not.
```

Comment:
I wonder if this is racy...i.e. can the state be changed after the parent does its check but before here?

Reply:
`state` could change, I suppose, but `teamname` is coming from `ownProps` which the parent passes directly, so I don't think assuming it's defined is a racy